controllers/release/release_adapter_test.go

```diff
@@ -0,0 +1,344 @@
+		Expect(strategy.Namespace).To(Equal(link.Spec.Target))
+	})
+
+	It("can return the target ReleaseLink from a given ReleaseLink", func() {
+		link, _ := adapter.getTargetReleaseLink()
+		Expect(reflect.TypeOf(link)).To(Equal(reflect.TypeOf(&appstudiov1alpha1.ReleaseLink{})))
+		Expect(link.Spec.Application).To(Equal("test-app"))
+		Expect(link.Spec.ReleaseStrategy).To(Equal(releaseStrategy.Name))
+	})
+
+	It("can mark the status for a given Release according to the pipelineRun status", func() {
+		pipelineRun := tekton.NewReleasePipelineRun("test-pipeline", testNamespace).AsPipelineRun()
+		pipelineRun.Name = "foo" // PipelineRun needs a name before registering the Release status
+		err := adapter.registerReleaseStatusData(pipelineRun, releaseStrategy)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(adapter.release.Status.StartTime).ToNot(BeNil())
+		pipelineRun.Status.MarkSucceeded("succeeded", "set it to succeeded")
+		Expect(pipelineRun.IsDone()).To(BeTrue())
+
+		err = adapter.registerReleasePipelineRunStatus(pipelineRun)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(release.HasSucceeded()).To(BeTrue())
+
+		// Double check that we still have a valid StartTime
+		Expect(adapter.release.Status.StartTime).ToNot(BeNil())
+
+		// Clear up previous condition so isDone returns false
+		release.Status.Conditions = []metav1.Condition{}
+		pipelineRun.Status.MarkFailed("failed", "set it to failed")
+		err = adapter.registerReleasePipelineRunStatus(pipelineRun)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(release.HasSucceeded()).To(BeFalse())
+	})
+
+	It("can add the release information to its status and mark it as Running", func() {
+		// The call should succeed if releaseStrategy or pipelineRun are nil
+		Expect(adapter.registerReleaseStatusData(nil, nil)).To(Succeed())
+
+		pipelineRun := tekton.NewReleasePipelineRun("test-pipeline", testNamespace).AsPipelineRun()
+		pipelineRun.Name = "new-name"
+		err := adapter.registerReleaseStatusData(pipelineRun, releaseStrategy)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(release.Status.ReleasePipelineRun).To(Equal(testNamespace + "/" + pipelineRun.Name))
+		Expect(release.Status.Conditions[0].Status == metav1.ConditionUnknown)
+		Expect(release.Status.Conditions[0].Reason == string(appstudiov1alpha1.ReleaseReasonRunning))
+	})
+
+	It("can update the Status of a release", func() {
+		patch := client.MergeFrom(adapter.release.DeepCopy())
+		adapter.release.Status.ReleasePipelineRun = "foo/test"
+
+		Expect(k8sClient.Status().Patch(ctx, release, patch)).Should(Succeed())
+		Expect(release.Status.ReleasePipelineRun).To(Equal("foo/test"))
+	})
+
+	It("can get an existing ApplicationSnapshot", func() {
+		snapshot, err := adapter.getApplicationSnapshot()
+		Expect(err).Should(Succeed())
+		Expect(reflect.TypeOf(snapshot)).To(Equal(reflect.TypeOf(&appstudioshared.ApplicationSnapshot{})))
+
+		Expect(k8sClient.Delete(ctx, snapshot)).Should(Succeed())
+
+		k8sClient.Get(ctx, types.NamespacedName{
```

Comment:
I thought I commented about this one. What for? It does nothing in the test.

Reply:
removed.